### ISSUE TITLE
The mock FileStream class does not handle shared file contents correctly.

### ISSUE THREAD

**Issue by BillArmstrong** [TestableIO/System.IO.Abstractions#1131]
The source code below should write the values 0-9 to the console. Instead it writes the value 0 to the console ten times.

The code behaves as expected when using the real file system.

```
using System.IO.Abstractions;
using System.IO.Abstractions.TestingHelpers;

//var fileSystem = new FileSystem();
var fileSystem = new MockFileSystem();

var filename = fileSystem.Path.GetTempFileName();
try
{
    using var file1 = fileSystem.FileStream.New(filename, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.ReadWrite);
    using var file2 = fileSystem.FileStream.New(filename, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
    var buffer = new byte[4];

    for (int ix = 0; ix < 10; ix++)
    {
        file1.Position = 0;
        file1.Write(BitConverter.GetBytes(ix));
        file1.Flush();

        file2.Position = 0;
        file2.Flush();
        file2.Read(buffer);
        Console.WriteLine(BitConverter.ToInt32(buffer));
    }
}
finally
{
    fileSystem.File.Delete(filename);
}
```

**Comment by vbreuss** [comment]
Thanks for reporting this issue. The root cause, seems to be, that the underlying MemoryStream is not updated when accessed from a parallel file handle.

**Comment by BekirK-C** [comment]
Hi, It looks like this issue has been fixed in [PR #625](https://github.com/Testably/Testably.Abstractions/pull/625), which has already been merged. Should this issue be closed or is there something else still pending? @vbreuss

**Comment by vbreuss** [comment]
@oni-shiro : It's not yet fixed here. I assigned it to you :-)

**Comment by oni-shiro** [comment]
I am bit confused with this issue, at base it is using the `FileStreamSystem` to write into file. Then should this change be part of this repo or not? 
`FileStremSystem` is coming from `Testably.Abstractions.FileSystem.Interface/FileSystemStream.cs` I guess.

**Comment by vbreuss** [comment]
Yes, you are right @oni-shiro, the `FileSystemStream` class is located [here](https://github.com/Testably/Testably.Abstractions/blob/main/Source/Testably.Abstractions.FileSystem.Interface/FileSystemStream.cs) and would need to be fixed there.

**Comment by mistial-dev** [comment]
The MockFileStream class does not handle shared file contents correctly when multiple streams are opened on the same file with FileShare.ReadWrite. Changes written through one stream were not visible to other streams, unlike real filesystem behavior.

Expected Behavior: When file1 writes data and flushes, file2 should be able to read the updated content.

Actual Behavior: file2 continued reading stale content (zeros) regardless of what file1 wrote.

Root Cause Analysis

1. Architecture Issue

Each MockFileStream instance creates its own independent MemoryStream (line 46 in MockFileStream.cs). While there is a shared MockFileData.Contents that represents the "file on disk," each stream operates on its own memory buffer.

2. Content Synchronization Problem

The synchronization between individual stream buffers and shared content is incomplete:

- Write → Shared: InternalFlush() correctly updated MockFileData.Contents when a stream flushed
- Shared → Read: Missing mechanism to refresh stream content from shared data before reads

3. Race Condition in Flush

The InternalFlush() method is called by both streams during Flush(), causing the second stream (which hadn't written anything) to overwrite the shared content with empty data:

file1.Write(data) → file1.Flush() → Updates shared content to [1,0,0,0]
file2.Flush() → Overwrites shared content to [] (empty)
file2.Read() → Reads empty data
